File: JJB/1202/최댓값_만들기(1).py
def solution(numbers):
    answer = 0
    
    first_max = max(numbers[0], numbers[1])
    second_max = min(numbers[0], numbers[1])
    
    for number in numbers[2:]:
        second_max = max(second_max, number)
        temp = max(first_max, second_max)
        second_max = min(first_max, second_max)
        first_max = temp
        
    answer = first_max * second_max
    
    return answer

File: JJB/Programmers/231202/최댓값_만들기(1).cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> numbers) 
{
    int answer = 0;
    
    int first_max = max(numbers[0], numbers[1]);
    int second_max = min(numbers[0], numbers[1]);
    
    for(const auto &number : vector<int>(numbers.begin()+2, numbers.end()))
    {
        second_max = max(second_max, number);
        int temp = max(first_max, second_max);
        second_max = min(first_max, second_max);
        first_max = temp;
    }
    
    answer = first_max * second_max;
    
    return answer;
}